pkg/ds/daemon_set.go

```diff
@@ -204,6 +207,13 @@ func (ds *daemonSet) WatchDesires(
 				err = ds.removePods()
 				if err != nil {
 					err = util.Errorf("Unable to remove pods from intent tree: %v", err)
+					// We must still try to add pods so that we can label the pods that
+					// potentially need to be scheduled because all updates are taken in
+					// only once
```

Comment:
It means that the WatchDiff function will calculate a diff and send out the changes, but it wouldn't ensure consistency between the daemon set labels and the pod labels because it sends a specific change only once.

So then we need to make sure that the pods that are scheduled for a replication will be labelled to belong to this daemon set so that a retryReplication would work.

Another way to handle the retryReplication is to do `removePods()`  and `addPods()`.